Feature/api english translation

7tobias opened the pull request:
## API Field Name Standardization
  This PR translates API field names from German to English to make the API more accessible to international developers while maintaining full backward compatibility.
 
  ## Changes
 
  ### Field Name Translations
 
  Renamed 18 fields across two main classes:
 
  **GeneticEnergyManagementParameters (5 fields):**
  - `pv_prognose_wh` → `pv_forecast_wh`
  - `strompreis_euro_pro_wh` → `electricity_price_per_wh`
  - `einspeiseverguetung_euro_pro_wh` → `feed_in_tariff_per_wh`
  - `preis_euro_pro_wh_akku` → `price_per_wh_battery`
  - `gesamtlast` → `total_load`
 
  **GeneticSimulationResult (13 fields):**
  - `Last_Wh_pro_Stunde` → `load_wh_per_hour`
  - `EAuto_SoC_pro_Stunde` → `ev_soc_per_hour`
  - `Einnahmen_Euro_pro_Stunde` → `revenue_per_hour`
  - `Gesamt_Verluste` → `total_losses`
  - `Gesamtbilanz_Euro` → `total_balance`
  - `Gesamteinnahmen_Euro` → `total_revenue`
  - `Gesamtkosten_Euro` → `total_costs`
  - `Kosten_Euro_pro_Stunde` → `costs_per_hour`
  - `Netzbezug_Wh_pro_Stunde` → `grid_consumption_wh_per_hour`
  - `Netzeinspeisung_Wh_pro_Stunde` → `grid_feed_in_wh_per_hour`
  - `Verluste_Pro_Stunde` → `losses_per_hour`
  - `akku_soc_pro_stunde` → `battery_soc_per_hour`
  - `Electricity_price` → `electricity_price` (already English)
 
  ### Backward Compatibility
 
  **Input:** Both English and German field names accepted via Pydantic `AliasChoices`
  **Output:** Both field names included in JSON via `@computed_field` properties
 
  No breaking changes - existing consumers continue to work.
 
  ### Currency-Neutral Field Names
 
  Removed currency-specific references from field names (e.g., `euro`) to support any currency. The system works with EUR, GBP, USD, SEK, CHF, etc. - all monetary values use the locally configured currency.
 
  Resolves #628

7tobias commented:
## UPDATE: This PR now contains BREAKING CHANGES

The field renaming has been updated to include **Python attribute names**, which means:

### Breaking Changes

**All genetic optimization API response fields are now in English.**

API consumers parsing response field names will need to update their code.

### Changed Fields

**`GeneticEnergyManagementParameters` (5 fields):**
- `pv_prognose_wh` → `pv_forecast_wh`
- `strompreis_euro_pro_wh` → `electricity_price_per_wh`
- `einspeiseverguetung_euro_pro_wh` → `feed_in_tariff_per_wh`
- `preis_euro_pro_wh_akku` → `price_per_wh_battery`
- `gesamtlast` → `total_load`

**`GeneticSimulationResult` (13 fields):**
- `Last_Wh_pro_Stunde` → `load_wh_per_hour`
- `EAuto_SoC_pro_Stunde` → `ev_soc_per_hour`
- `Einnahmen_Euro_pro_Stunde` → `revenue_per_hour`
- `Gesamt_Verluste` → `total_losses`
- `Gesamtbilanz_Euro` → `total_balance`
- `Gesamteinnahmen_Euro` → `total_revenue`
- `Gesamtkosten_Euro` → `total_costs`
- `Kosten_Euro_pro_Stunde` → `costs_per_hour`
- `Netzbezug_Wh_pro_Stunde` → `grid_consumption_wh_per_hour`
- `Netzeinspeisung_Wh_pro_Stunde` → `grid_feed_in_wh_per_hour`
- `Verluste_Pro_Stunde` → `losses_per_hour`
- `akku_soc_pro_stunde` → `battery_soc_per_hour`

### Input Compatibility (Backward Compatible)

Both German and English field names work as **input**:

```python
# Both accepted:
{"pv_prognose_wh": [100]}   #  German
{"pv_forecast_wh": [100]}   #  English
```

### Output Format (Breaking Change)

Only English field names in **output**:

```python
# Response is always:
{"pv_forecast_wh": [100], ...}  # English only
```

### Migration Required

```python
# Before (will break):
result['pv_prognose_wh']      #  KeyError
result['Gesamtkosten_Euro']   #  KeyError

# After (correct):
result['pv_forecast_wh']      #  Works
result['total_costs']         #  Works
```

###  Rationale

For an international open-source project, English API field names are the standard. This makes the project accessible to users worldwide (Sweden, Switzerland, UK, etc.) while maintaining input compatibility with existing German names.

7tobias commented:
@NormannK Sorry it took me a while to get back to this. The failing test is fixed now — could you please reopen this PR? I can't reopen it myself since the branch was updated after it was closed.

What changed:
 - Fixed the test failure: the visualization was still being fed the raw German-keyed simulation dict (KeyError: load_wh_per_hour')
 - Rebased on current main (conflicts with the genetic optimizer rework resolved)
 - Completed backward compatibility: all 14 renamed result fields are re-emitted under their deprecated German names, marked deprecated in the OpenAPI schema
 - Documented that monetary values are whole currency units, never cents - addressing @Xexanos' remark

The full test suite passes. If reopening isn't possible, I'll open a fresh PR from the same branch.

NormannK commented:
looking good. i also saw "restwert_akku" and "gesamtbilanz"

7tobias commented:
Thanks for the approval! I pushed one more commit addressing your remark: renamed `restwert_akku` and `gesamtbilanz` plus everything else a full sweep turned up — remaining internal German identifiers (`eigenverbrauch`, `optimierung_ems`, the `bilanz`/`verluste`/`nebenbedingung` dict keys) and hardcoded € in chart labels and schema descriptions, which are now currency-neutral to match the API description.

The input fields `pv_akku`/`eauto` and the legacy `/strompreis`/`/gesamtlast` endpoints I'd leave for a follow-up PR, since renaming those touches the request schema — happy to do that next if you're interested.

NormannK commented:
@b0661 are you ok with all the deprecated names?

b0661 commented:
> @b0661 are you ok with all the deprecated names?

My only nitpick is currency. I would call it amount (short amt.), as this is what is used in banking. E.g. amount/kWh.

NormannK commented:
@7tobias i tend to agree with b0661. could you add it to this PR or should we make a new one and accept this?

7tobias commented:
Good suggestion. I will add it.

7tobias commented:
Done - using amount now.

b0661 commented on src/akkudoktoreos/core/ems.py:
```diff
@@ -188,170 +193,205 @@ def _run(
-            start_datetime (DateTime, optional): The starting timestamp
-                of the energy management run. Defaults to the current datetime
-                if not provided.
-            mode (EnergyManagementMode, optional): The management mode to use. Must be one of:
```
This looks weird. Please double check that this is not destroying something.

7tobias commented:
@b0661 Double checked: this branch's `ems.py` is identical to main except the two renamed `optimize_ems` calls (`git diff main -- src/akkudoktoreos/core/ems.py`) — your async refactor is untouched, tests and CI are green.

The big diff comes from the merge commits I used to keep the branch updated; the actual PR diff vs main is small. Happy to rebase into a few clean commits and force-push if you prefer.

Commit:
feat: rename genetic optimization API fields to English

Rename the German field names of the genetic optimization API to
English with full backward compatibility:

- English names are canonical and documented in the OpenAPI schema;
  the German names are still accepted on input via validation aliases
  and re-emitted in responses as deprecated computed fields
- fix visualization receiving the raw German-keyed simulation dict
  (KeyError: load_wh_per_hour)
- rename internal German identifiers (optimize_ems, total_balance,
  battery_residual_value, self_consumption, extra_data keys)
- use amount instead of currency/euro in chart labels and schema
  descriptions; document currency handling (whole currency units,
  never cents) in the API description
- regenerate openapi.json and generated docs

7tobias commented:
Rebased to main
